quinn-proto/src/endpoint.rs

```diff
@@ -75,6 +84,7 @@ impl Endpoint {
         config: Arc<EndpointConfig>,
         server_config: Option<Arc<ServerConfig>>,
         allow_mtud: bool,
+        transmit_queue_contents_len: Arc<AtomicUsize>,
```

Comment:
This is a breaking change, and hence defeats the purpose of the backport. A better solution might be to add a getter on `proto::Endpoint` for the current queue length, which could then be called from the `quinn::Endpoint`. This also avoids needing shared ownership of an atomic.

Reply:
@Ralith Can you elaborate how this is a breaking change? Older quinn crate with the newer quinn-proto version? Adding a getter at the proto::Endpoint cannot solve the issue as we are doing the suppression at proto::Endpoint not at quinn::Endpoint. Am I missing something?

Reply:
Maybe provide a getter to get the Arc<AtomicUsize> at proto::Endpoint? If so, that can work. For the breaking change, I guess you are referring to the proto::Endpoint::new interface change?

Reply:
Changing the signature of a public function is always a breaking change, and hence cannot be made in a patch release.

A getter accessing the current total size of the `proto::Endpoint` transmit queue would be much simpler than sharing an atomic.

Reply:
Got it. We will need to setters (increment and decrement)

Reply:
Rather than separate increment/decrement (which is error-prone), a setter for a single absolute count that the -proto layer should consider in addition to its internal count would be simpler and more robust.

Reply:
What should we call that counter?

Reply:
How about connection_transmit_queue_contents_len?

Reply:
Perhaps `socket_buffer_fill`? Doesn't matter too much, since this is only a backport.

Reply:
Done

Reply:
Doesn't look like you've pushed?

Reply:
ahh -- sorry,  re-pushed to the right branch!